scio-core/src/main/scala/com/spotify/scio/values/SCollectionWithSideInput.scala

```diff
@@ -82,6 +90,101 @@ class SCollectionWithSideInput[T] private[values] (
+    val o = coll
+      .pApply(parDo(new FlatMapFnWithResource(resource, resourceType, f)))
+      .setCoder(CoderMaterializer.beam(context, Coder[U]))
+    new SCollectionWithSideInput[U](o, sides)
+  }
+
+  /** [[map]] with additional resource arguments. */
+  def mapWithResource[R, U: Coder](resource: => R, resourceType: ResourceType)(
+    fn: (R, T, SideInputContext[T]) => U
+  ): SCollectionWithSideInput[U] = {
+    val o = coll
+      .pApply(parDo(new MapFnWithWithResource(resource, resourceType, fn)))
+      .setCoder(CoderMaterializer.beam(context, Coder[U]))
+    new SCollectionWithSideInput[U](o, sides)
+  }
+
+  /** [[collect]] with additional resource arguments. */
+  def collectWithResource[R, U: Coder](resource: => R, resourceType: ResourceType)(
+    pfn: PartialFunction[(R, T, SideInputContext[T]), U]
+  ): SCollectionWithSideInput[U] = {
+    val o = coll
+      .pApply(parDo(new CollectFnWithResource(resource, resourceType, pfn)))
+      .setCoder(CoderMaterializer.beam(context, Coder[U]))
+    new SCollectionWithSideInput[U](o, sides)
+  }
+
+  /** [[SCollection.batch]] that retains [[SideInput]]. */
+  def batch(
+    batchSize: Long,
+    maxLiveWindows: Int = BatchDoFn.DEFAULT_MAX_LIVE_WINDOWS
+  ): SCollectionWithSideInput[Iterable[T]] =
+    new SCollectionWithSideInput[Iterable[T]](coll.batch(batchSize, maxLiveWindows), sides)
+
+  /** [[SCollection.batchByteSized]] that retains [[SideInput]]. */
+  def batchByteSized(
+    batchByteSize: Long,
+    maxLiveWindows: Int = BatchDoFn.DEFAULT_MAX_LIVE_WINDOWS
+  ): SCollectionWithSideInput[Iterable[T]] =
+    batchWeighted(batchByteSize, ScioUtil.elementByteSize(context), maxLiveWindows)
+
+  /** [[SCollection.batchWeighted]] that retains [[SideInput]]. */
+  def batchWeighted(
+    batchWeight: Long,
+    cost: T => Long,
+    maxLiveWindows: Int = BatchDoFn.DEFAULT_MAX_LIVE_WINDOWS
+  ): SCollectionWithSideInput[Iterable[T]] =
+    new SCollectionWithSideInput[Iterable[T]](
+      coll.batchWeighted(batchWeight, cost, maxLiveWindows),
+      sides
+    )
+
+  /** [[SCollection.withPaneInfo]] that retains [[SideInput]]. */
+  def withPaneInfo: SCollectionWithSideInput[(T, PaneInfo)] =
+    new SCollectionWithSideInput(coll.withPaneInfo, sides)
+
+  /** [[SCollection.withTimestamp]] that retains [[SideInput]]. */
+  def withTimestamp: SCollectionWithSideInput[(T, Instant)] =
+    new SCollectionWithSideInput(coll.withTimestamp, sides)
+
+  /** [[SCollection.withTimestamp]] that retains [[SideInput]]. */
+  def withWindow[W <: BoundedWindow: Coder]: SCollectionWithSideInput[(T, W)] =
+    new SCollectionWithSideInput(coll.withWindow, sides)
+
+  /** [[SCollection.withFixedWindows]] that retains [[SideInput]]. */
+  def withFixedWindows(
+    duration: Duration,
+    offset: Duration = Duration.ZERO,
+    options: WindowOptions = WindowOptions()
+  ): SCollectionWithSideInput[T] =
+    new SCollectionWithSideInput(coll.withFixedWindows(duration, offset, options), sides)
```

Comment:
I think we probably don't need these methods, because the user can just write `data.withPaneInfo.withSideInputs` rather than `data.withSideInputs.withPaneInfo` and end up with the same result, right? 🤔

Reply:
yes you're right, i didn't know if there was a use case for someone who already has a SCollectionWithSIdeInput then later wanted to get pane info. I'll remove them.